### PR TITLE
Effect: Make .transfer() callback optional

### PR DESCRIPTION
**Pull request by jzaefferer**
Adds two clunky tests to at least check that no exception is thrown.

Fixes #12223

### CODE REVIEW

File: tests/unit/effects/core.js [hunk 1]
```diff
@@ -258,6 +258,29 @@ test( "createPlaceholder: preserves layout affecting properties", function() {
...
+	}, 25 );
+} );
+
+asyncTest( "transfer() with callback", function() {
+	expect( 0 );
+	$( "#elem" ).transfer( {
+		to: ".animateClass",
+		duration: 1
+	}, function() {
+		start();
```
Comment (scottgonzalez) [inline tests/unit/effects/core.js]: Seems like you could add an `ok( true, "callback invoked" )` to at least have an assertion here.

File: tests/unit/effects/core.js [hunk 2]
```diff
@@ -258,6 +258,29 @@ test( "createPlaceholder: preserves layout affecting properties", function() {
...
+asyncTest( "transfer() without callback", function() {
+	expect( 0 );
+	$( "#elem" ).transfer( {
+		to: ".animateClass",
+		duration: 1
+	} );
+	setTimeout( function() {
+		start();
+	}, 25 );
+} );
```
Comment (scottgonzalez) [inline tests/unit/effects/core.js]: Perhaps add a comment explaining what we're testing here.

**Comment by jzaefferer** [comment]
Updated

File: tests/unit/effects/core.js [hunk 3]
```diff
@@ -258,6 +258,32 @@ test( "createPlaceholder: preserves layout affecting properties", function() {
...
+
+asyncTest( "transfer() with callback", function() {
+	expect( 0 );
```
Comment (scottgonzalez) [inline tests/unit/effects/core.js]: Is this actually passing? Should be `expect( 1 )`.

Comment (jzaefferer) [inline tests/unit/effects/core.js]: Yeah, that was stupid. Fixed.

**Comment by scottgonzalez** [comment]
Looks good.

**Pull request merged**